compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -2918,7 +2914,25 @@ static void generateArrayElementStore(TR::Node* node, TR::Register* addressReg,
 static void generateArrayElementLoad(TR::Node* node, TR::Register* valueReg, uint8_t size, TR::Register* addressReg, int32_t index, TR::CodeGenerator* cg)
    {
    TR::InstOpCode::Mnemonic loadOpCode;
-   if (valueReg->getKind() == TR_FPR)
+   if (valueReg->getKind() == TR_VRF)
+      {
+      switch (size)
+         {
+         case 16:
+            loadOpCode  = TR::InstOpCode::MOVDQURegMem;
+            break;
+         case 32:
+            loadOpCode  = TR::InstOpCode::VMOVDQUYmmMem;
+            break;
+         case 64:
+            loadOpCode  = TR::InstOpCode::VMOVDQUZmmMem;
+            break;
+         default:
+            TR_ASSERT(0, "%s: Unsupported size %u for TR_VRF registers\n", __FUNCTION__, size);
```

Comment:
I'd say if you add or modify any of the `TR_ASSERT`s in this PR then you should change them to `TR_ASSERT_FATAL`s instead to be actually useful in practice.  This applies not just here but to all the asserts you touch in this PR (I won't put a comment on each).

Reply:
Addressed  in [d2c22d0](https://github.com/eclipse/omr/compare/d2c22d0bf1d27788f27712939d4e22e79ba06e0c..24c7d3f9ddf90cc7165e3bacb2ca8d63d053c0d7)